code/modules/atmospherics/gasmixtures/gas_mixture.dm

```diff
@@ -70,7 +70,7 @@ GLOBAL_LIST_INIT(gaslist_cache, init_gaslist_cache())
 /datum/gas_mixture/proc/garbage_collect(list/tocheck)
 	var/list/cached_gases = gases
 	for(var/id in (tocheck || cached_gases))
-		if(QUANTIZE(cached_gases[id][MOLES]) <= 0 && QUANTIZE(cached_gases[id][ARCHIVE]) <= 0)
+		if(QUANTIZE(cached_gases[id][MOLES]) < MINIMUM_MOLE_COUNT && QUANTIZE(cached_gases[id][ARCHIVE]) < MINIMUM_MOLE_COUNT)

```

Comment:
I'm interested to see how this behaves ingame, because this might break a whole lot of things.
For context, let's look at the comment left on QUANTIZE so we can better understand the goal of this function.

`#define QUANTIZE(variable)		(round(variable,0.0000001))/*I feel the need to document what happens here. Basically this is used to catch most rounding errors, however it's previous value made it so that
															once gases got hot enough, most procedures wouldnt occur due to the fact that the mole counts would get rounded away. Thus, we lowered it a few orders of magnititude */`

So far as I know the use of quantize here is similar to changing the 0 to 0.0000001. You've raised this number to MINIMUM_MOLE_COUNT, which has a value of 0.01. 

If I understand the system correctly this will break fusion and kill gasses at high temperatures. I sympathize with the goal of removing those annoying 0kpa gasses from pipes but this is the wrong way to go about it.

Reply:
Actually QUANTIZE used only in plasmafire
https://github.com/tgstation/tgstation/blob/51afca0dfc58bc50179b90db24c85b51b47601e4/code/modules/atmospherics/gasmixtures/reactions.dm#L179-L239
and garbage_collect()
https://github.com/tgstation/tgstation/blob/51afca0dfc58bc50179b90db24c85b51b47601e4/code/modules/atmospherics/gasmixtures/gas_mixture.dm#L70-L74


The MINIMUM_MOLE_COUNT maybe need make some smaller, from 0.01 to 0.001 or 0.0001, this need testing.

changing the 0 to 0.0000001 is some weird. This make imposible to delete gas from mix if it dont grabbed complitely.

Now i think about add some like MINIMUM_MOLE_MOVE = 0.01 or 0.001 to all places where moles moved partially, so no more pumps try move 0.00001 moles of gas infinitely.

Now i think about make atmos machinery work more similar to pipes. Now pipeline share gas from self pipe and all connected things by reconcile_air()
https://github.com/tgstation/tgstation/blob/51afca0dfc58bc50179b90db24c85b51b47601e4/code/modules/atmospherics/machinery/datum_pipeline.dm#L211-L264

I think about make machinery gases absorbed to pipeline so no more need reconcile_air() and machinery work directly to pipeline gas pool. 
As i see volume_pump already can use this type system
https://github.com/tgstation/tgstation/blob/51afca0dfc58bc50179b90db24c85b51b47601e4/code/modules/atmospherics/machinery/components/binary_devices/volume_pump.dm#L53-L86
 since it use `remove_ratio(transfer_rate/air1.volume)` where `air1.volume` = 200 now and `transfer_rate` max is 200.
When `air1.volume` merged in pipeline gas pull it still take only 200 liters from this big pipeline gas pull.